providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdCache.java

```diff
@@ -11,7 +11,7 @@
  * Exposes caching mechanism for flag evaluations.
  */
 public class FlagdCache {
-    private Map<String,ProviderEvaluation<Value>> store;
+    private Map<String,ProviderEvaluation<? extends Object>> store;
```

Comment:
I would expect that all elements stored can be casted to `Value`
```suggestion
    private Map<String,ProviderEvaluation<? super Value>> store;
```

Reply:
`Value` is actually a wrapper type that represents a JSON value (including JSON structures). We could convert all our resolved values to this, but it's not necessary. I've chosen `? extends Object` because it's the most "honest" type for this - gRPC can return Bools, Floats, Longs, Strings, Structs here, so Object makes sense to me - but maybe I'm missing something.

Reply:
Looking into the code, the type held by `ProviderEvaluation` is never checked. Your suggestion is valid, sorry for the noise 😇

Reply:
Glad to have a check from you!